Userland/Libraries/LibVirtGPU/CommandBufferBuilder.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2022, Sahan Fernando <sahan.h.fernando@gmail.com>
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringView.h>
+#include <Kernel/API/VirGL.h>
+#include <sys/ioctl_numbers.h>
+
+#include "CommandBufferBuilder.h"
+#include "VirGLProtocol.h"
+
+constexpr int DRAWTARGET_WIDTH = 500;
+constexpr int DRAWTARGET_HEIGHT = 500;
+
+static u32 encode_command(u32 length, u32 mid, Protocol::VirGLCommand command)
+{
+    u32 command_value = to_underlying(command);
+    return (length << 16) | ((mid & 0xff) << 8) | (command_value & 0xff);
+};
+
+class CommandBuilder {
+public:
+    CommandBuilder() = delete;
+    CommandBuilder(Vector<u32>& buffer, Protocol::VirGLCommand command, u32 mid)
+        : m_buffer(buffer)
+        , m_start_offset(buffer.size())
+        , m_command(command)
+        , m_command_mid(mid)
+    {
+        m_buffer.append(0);
+    }
+    void appendu32(u32 value)
+    {
+        VERIFY(!m_finalized);
+        m_buffer.append(value);
+    }
```

Comment:
These method declarations in this file should be separated by a newline after the `}`